recorder/src/main/cpp/perf_ctx.cc

```diff
@@ -103,10 +103,15 @@ void PerfCtx::ThreadTracker::exit(PerfCtx::TracePt pt) throw (IncorrectEnterExit
         }
     }
 
+    if (actual_stack.size() == 0) {
```

Comment:
This is the exit routine. Stack-size is zero when you call ctx.end() before calling ctx.begin().

This case: https://github.com/Flipkart/fk-prof/pull/106/files/5e12a06e8c58173a224f9fa218f26846fc2e7f37#diff-6e78670c35d971e61e3eef5d534a87c5R431